File: src/main/java/org/athenian/ListSum.java
package org.athenian;

import java.util.List;

public class ListSum {

    public static int listSumLoop(List<Integer> list) {
        int sum = 0;
        for (Integer i : list)
            sum += i;
        return sum;
    }

    public static int listSumRecursion(List<Integer> list) {
        if (list.isEmpty())
            return 0;

        return list.get(0) + listSumRecursion(list.subList(1, list.size()));
    }

    public static int listSumStream(List<Integer> list) {
        return list
                .stream()
                .mapToInt(i -> i)
                .sum();
    }
}

File: src/main/kotlin/org/athenian/ListSum.kt
package org.athenian


fun listSumLoop(list: List<Int>): Int {
    var sum = 0
    list.indices.forEach { sum += list[it] }
    return sum
}

fun listSumRecursion(list: List<Int>): Int =
    when {
        list.isEmpty() -> 0
        else -> list[0] + listSumRecursion(list.slice(1..list.size - 1))
    }

fun listSumStream(list: List<Int>) = list.sum()